libafl/src/inputs/mod.rs

```diff
@@ -72,17 +87,103 @@ pub trait Input: Clone + Serialize + serde::de::DeserializeOwned + Debug {
         Ok(postcard::from_bytes(&bytes)?)
     }
 
+    /// Serializes this input to the dynamic serialisation format to pass between different fuzzers
+    fn serialize_dynamic(&self, buf: &mut Vec<u8>) -> Result<(), postcard::Error> {
+        buf.extend_from_slice(postcard::to_allocvec(Self::NAME)?.as_slice());
+        buf.extend_from_slice(postcard::to_allocvec(self)?.as_slice());
+        Ok(())
+    }
+
+    /// Deserializes this input type from the dynamic serialization format, if possible
+    fn deserialize_dynamic(
+        buf: &[u8],
+    ) -> Result<Option<Self>, <&mut Deserializer<Slice> as serde::de::Deserializer>::Error> {
+        convert_named(buf)
+    }
```

Comment:
These two methods can be overriden, so it's possible to serialize context and pass it at the end of buf here; just append the context to the end of the message. If the target already has it, it can opt to simply not deserialize it.

Reply:
I'm not sure how big context is, but if it's reasonably sized this could be effective.

Reply:
context doesn't need to be serialized, but it is needed to do the serialization as it contains some info to convert the AST to bytes

Reply:
It doesn't. To serialize with serde there is no need of it, while it is needed to convert the AST stored in the input to bytes. You see it only in the harness.

Reply:
That's also why I was proposing a manager that takes two lambdas to convert an input to bytes and back. We can probably add a generic parameter to serialize_dynamic or similar stuffs

Reply:
Also EncodedInput has something similar, as it is an array of u32 id representing tokens and the mapping id -> token is ofc not stored in the input itself so to convert to bytes you need to call https://github.com/AFLplusplus/LibAFL/blob/main/libafl/src/inputs/encoded.rs#L75

Reply:
Yes

Reply:
But I see why it is convenient to do in the receiver, we can reuse the NewTestcase event and avoid double memory usage

Reply:
I have to think about it, but I don't see other solutions, the tokens map in the EncodedInput decoder can be large even gigabytes when using a huge initial corpus